### PR TITLE
KRAMDOWN_NO_TARGETS: let xml2rfc fill in targets for RFC and I-D

### PR DESCRIPTION
**Pull request by cabo**
refactor reference mangling to use REXML

### CODE REVIEW

File: lib/kramdown-rfc2629.rb [hunk 1]
```diff
@@ -839,10 +842,32 @@ def convert_img(el, indent, opts) # misuse the tag!
             # end
             to_insert = get_and_cache_resource(url, fn.gsub('/', '_'), ttl)
             to_insert.scrub! rescue nil # only do this for Ruby >= 2.1
-            to_insert.gsub!(%r{target="https?://www.ietf.org/internet-drafts/},
-                            %{target="https://www.ietf.org/archive/id/}) if t == "I-D"
+
+            begin
```
Comment (martinthomson) [inline lib/kramdown-rfc2629.rb:846]: I would factor this block out into a new function.

File: lib/kramdown-rfc2629.rb [hunk 2]
```diff
@@ -839,10 +842,32 @@ def convert_img(el, indent, opts) # misuse the tag!
...
+              d = REXML::Document.new(to_insert)
+              d.xml_decl.nowrite
+              d.root.attributes["anchor"] = anchor
+              if t == "RFC" or t == "I-D"
+                if KRAMDOWN_NO_TARGETS
```
Comment (martinthomson) [inline lib/kramdown-rfc2629.rb:851]: WFM

**Review by martinthomson — APPROVED** [review]
This works nicely.

**Comment by cabo** [comment]
Damn the torpedoes...

**Pull request merged**